source/server/listener_manager_impl.h

```diff
@@ -194,7 +194,9 @@ class ListenerImpl : public Listener,
   Network::FilterChainFactory& filterChainFactory() override { return *this; }
   Network::ListenSocket& socket() override { return *socket_; }
   bool bindToPort() override { return bind_to_port_; }
-  Ssl::ServerContext* sslContext() override { return ssl_context_.get(); }
+  Ssl::ServerContext* sslContext() override {
```

Comment:
Who calls this?  Is the 0th ssl context the right one to return in this case?  Maybe this function should go away, or start returning a const std::vector<Ssl::ServerContextPtr>& ?

Reply:
This is called when creating listener to get initial `SSL_CTX`. 0-th element is as good as any, really, since we're going to update it later anyway.

Returning `std::vector<Ssl::ServerContextPtr>` doesn't make much sense, since caller doesn't have any insight (at least not yet) about which context to chose.

Reply:
I think this should be renamed to "sslDefaultContext()".

Reply:
Renamed to `defaultSslContext()`.